### DISCUSSION TITLE
[processor] Is it possible to generate metrics from logs?

### DISCUSSION THREAD

**Discussion by zvrastil** [open-telemetry/opentelemetry-collector-contrib#10583] [Q&A] [answered]
HIs it possible to write a custom processor that will process logs and generate new metrics from those logs? We have existing system that puts a lot of interesting metadata in some logs. We'd like to generate metrics from those logs on-the-fly. 

Custom processor looks like a good way to do it, but it looks like processors are confined to their pipelines, so log processor can only mutate logs, but not create metrics from it. Or do I understand it wrong?

**Comment by garry-cairns** [comment, marked as answer]
I'm not a contributor but I think if you look at the spanmetrics processor it does something similar but with spans, and may be a good starting point for what you suggest.

**Reply by zvrastil** [reply]
Yes, that looks exactly as what I'm trying to achieve. So trace/log processor can feed metrics into the exporter of the metrics pipeline, as long as the metrics pipeline exists and its exporter is given to the processor as part of the configuration.

Thanks a lot!

**Comment by wbollock** [comment]
https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/connector/signaltometricsconnector is the best solution for anyone else finding this thread in the future
